src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,169 @@
+#include "depthai/depthai_wrapper.hpp"
+
+namespace DepthAI
+{
+
+DepthAI::DepthAI(std::string usb_device, std::string config_file, bool usb2_mode) :
+      Device(usb_device, usb2_mode)
+{
+
+    std::ifstream file(config_file);
+    std::ostringstream file_stream;
+    file_stream << file.rdbuf();
+    std::string config_str = file_stream.str();
+
+    pipeline_ = this->create_pipeline(config_str);
+    config_json_ = nlohmann::json::parse(config_str);
+
+    this->set_resolution();
+    this->create_frame_holders();
+
+    width_map_[720] = 1280;
+    width_map_[800] = 1280;
+    width_map_[400] = 640;
+}
+
+void DepthAI::set_resolution()
+{
+
+    if (config_json_.contains("camera"))
+    {
+        auto& camera_conf_obj = config_json_.at("camera");
+        if (camera_conf_obj.contains("rgb"))
+        {
+            auto& rgb_camera_conf_obj = camera_conf_obj.at("rgb");
+            rgb_height_ = rgb_camera_conf_obj.at("resolution_h").get<int32_t>();
+            // _rgb_width = width_map_[_rgb_height];
+        }
+        else
+        {
+            rgb_height_ = 3040;
+            rgb_width_ = 4056;
+        }
+
+        if (camera_conf_obj.contains("mono"))
+        {
+            auto& mono_camera_conf_obj = camera_conf_obj.at("mono");
+            mono_height_ = mono_camera_conf_obj.at("resolution_h").get<int32_t>();
+            mono_width_ = width_map_[mono_height_];
+        }
+        else
+        {
+            mono_height_ = 720;
+            mono_width_ = 1280;
+        }
+    }
+    else
+    {
+        rgb_height_ = 3040;
+        rgb_width_ = 4056;
+        mono_height_ = 720;
+        mono_width_ = 1280;
+    }
+}
+
+void DepthAI::create_frame_holders()
+{
+
+    if (config_json_.contains("streams"))
+    {
+        for (auto it : config_json_.at("streams"))
+        {
+            // "metaout"
+            if (it.is_string())
+            {
+                if (it.get<std::string>() == "metaout")
+                    continue;
```

Comment:
Are we skipping this because we don't support it yet? Also, remember `continue` skips over all the other steps in this iteration. Is that what you're trying to do here?

Reply:
Yes